apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4CoverageResourceProvider.java

```diff
@@ -64,38 +65,35 @@ public final class R4CoverageResourceProvider implements IResourceProvider {
   /** The entity manager. */
   private EntityManager entityManager;
   /** The Metric registry. */
-  private MetricRegistry metricRegistry;
+  private final MetricRegistry metricRegistry;
   /** The Loaded filter manager. */
-  private LoadedFilterManager loadedFilterManager;
+  private final LoadedFilterManager loadedFilterManager;
 
   /**
-   * Sets the {@link #entityManager}.
+   * Instantiates a new {@link R4CoverageResourceProvider}.
    *
-   * @param entityManager a JPA {@link EntityManager} connected to the application's database
-   */
-  @PersistenceContext
-  public void setEntityManager(EntityManager entityManager) {
-    this.entityManager = entityManager;
-  }
-
-  /**
-   * Sets the {@link #metricRegistry}.
+   * <p>Spring will wire this class during the initial component scan, so this constructor should
+   * only be explicitly called by tests.
    *
-   * @param metricRegistry the {@link MetricRegistry} to use
+   * @param metricRegistry the metric registry
+   * @param loadedFilterManager the loaded filter manager
    */
-  @Inject
-  public void setMetricRegistry(MetricRegistry metricRegistry) {
+  public R4CoverageResourceProvider(
+      MetricRegistry metricRegistry, LoadedFilterManager loadedFilterManager) {
+    requireNonNull(metricRegistry);
+    requireNonNull(loadedFilterManager);
     this.metricRegistry = metricRegistry;
+    this.loadedFilterManager = loadedFilterManager;
```

Comment:
This could be "simplified" through Lombok's `@RequiredArgsConstructor`, I know where doing `requireNonNull` here, but being that this is constructed by the DI container, I don't believe they can be null, except in tests.

Reply:
If the beans are changed such that they cannot be found at DI time, they'll be null. (I think Spring will also throw a slightly less helpful exception if it can't find a dependency at injection time) It's not super likely but catches issues with the bean setup being changed.